src/main/java/org/jenkinsci/plugins/codesonar/api/CodeSonarHubAnalysisDataLoader.java

```diff
@@ -0,0 +1,238 @@
+    private CodeSonarAnalysisWarningCount activeWarningsCount;
+    private CodeSonarAnalysisWarningCount newWarningsCount;
+    private Analysis analysisViewActive;
+    private Analysis analysisViewNew;
+    private Metrics metrics;
+    private Procedures procedures;
+    private ProcedureMetric procedureWithMaxCyclomaticComplexity;
+    private CodeSonarAlertCounter alertCounter;
+    private Map<Integer,Long> numberOfWarningsAboveThreshold = new HashMap<>();
+    //------------------------------------------------------------
+    
+    public CodeSonarHubAnalysisDataLoader(HttpService httpService, CodeSonarHubInfo hubInfo, URI baseHubUri, Long analysisId, String visibilityFilter, String newWarningsVisibilityFilter) {
+        this.services = new CodeSonarServices(httpService, hubInfo);
+        this.hubInfo = hubInfo;
+        this.baseHubUri = baseHubUri;
+        this.analysisId = analysisId;
+        this.visibilityFilter = visibilityFilter;
+        this.newWarningsVisibilityFilter = newWarningsVisibilityFilter;
+    }
+    
+    public URI getBaseHubUri() {
+        return baseHubUri;
+    }
+    
+    protected Long getAnalysisId() {
+        return analysisId;
+    }
+
+    protected String getVisibilityFilter() {
+        return visibilityFilter;
+    }
+
+    protected String getNewWarningsVisibilityFilter() {
+        return newWarningsVisibilityFilter;
+    }
+    
+    protected Analysis getLegacyAnalysisViewActive() throws IOException {
+        LOGGER.log(Level.INFO, "AnalysisViewActive not already set, loading from corresponding legacy service");
+        services.getAnalysisService().setVisibilityFilter(visibilityFilter);
+        return services.getAnalysisService().getAnalysisFromUrlWarningsByFilter(getBaseHubUri(), getAnalysisId());
+    }
+
+    protected Analysis getLegacyAnalysisViewNew() throws IOException {
+        LOGGER.log(Level.INFO, "AnalysisViewNew not already set, loading from corresponding legacy service");
+        services.getAnalysisService().setVisibilityFilter(newWarningsVisibilityFilter);
+        return services.getAnalysisService().getAnalysisFromUrlWithNewWarnings(getBaseHubUri(), getAnalysisId());
+    }
+
+    protected Procedures getLegacyProcedures() throws IOException {
+        LOGGER.log(Level.INFO, "Procedures not already set, loading from corresponding legacy service");
+        URI proceduresUri = services.getProceduresService().getProceduresUriFromAnAnalysisId(baseHubUri, String.valueOf(analysisId));
+        return services.getProceduresService().getProceduresFromUri(proceduresUri);
+    }
+    
+    protected Metrics getLegacyMetrics() throws IOException {
+        LOGGER.log(Level.INFO, "Metrics not already set, loading from corresponding legacy service");
+        URI metricsUri = services.getMetricsService().getMetricsUriFromAnAnalysisId(getBaseHubUri(), String.valueOf(analysisId));
+        return services.getMetricsService().getMetricsFromUri(metricsUri);
+    }
```

Comment:
Is this method used anywhere?  Should it be?  How did it get here?

Reply:
This method was somehow used by those graphs that the plugin used to create, about an year ago.